File: src/main/java/com/oshi/libgenericsettings/viewholder/SettingsTitleSubtitleExtraViewHolder.java
package com.oshi.libgenericsettings.viewholder;

import android.databinding.ViewDataBinding;

import com.oshi.libgenericsettings.BR;
import com.oshi.libgenericsettings.data.TitleSubtitleExtraData;
import com.oshi.libgenericsettings.presenter.ISettingsPresenter;

/**
 * Created by udioshi on 9/4/2017.
 */

public class SettingsTitleSubtitleExtraViewHolder extends BaseSettingsViewHolder<TitleSubtitleExtraData> {

    public SettingsTitleSubtitleExtraViewHolder(ViewDataBinding viewDataBinding) {
        super(viewDataBinding);
    }

    @Override
    public void onBind(TitleSubtitleExtraData data, ISettingsPresenter presenter, int position) {
        viewDataBinding.setVariable(BR.titleSubtitleExtraData, data);
        viewDataBinding.setVariable(BR.itemPosition, position);
        viewDataBinding.setVariable(BR.settingsPresenter, presenter);
        viewDataBinding.executePendingBindings();
    }
}

File: src/main/java/com/oshi/libgenericsettings/viewholder/SettingsTitleSubtitleExtraViewHolder.kt
package com.oshi.libgenericsettings.viewholder

import android.databinding.ViewDataBinding
import com.oshi.libgenericsettings.BR
import com.oshi.libgenericsettings.data.TitleSubtitleExtraData
import com.oshi.libgenericsettings.presenter.ISettingsPresenter


class SettingsTitleSubtitleExtraViewHolder(viewDataBinding: ViewDataBinding) : BaseSettingsViewHolder<TitleSubtitleExtraData>(viewDataBinding) {

    override fun onBind(data: TitleSubtitleExtraData, presenter: ISettingsPresenter, position: Int) {
        viewDataBinding.setVariable(BR.titleSubtitleExtraData, data)
        viewDataBinding.setVariable(BR.itemPosition, position)
        viewDataBinding.setVariable(BR.settingsPresenter, presenter)
        viewDataBinding.executePendingBindings()
    }

}